java/ql/test/experimental/query-tests/security/CWE-400/ThreadResourceAbuse.java

```diff
@@ -105,4 +149,47 @@ protected void doHead(HttpServletRequest request, HttpServletResponse response)
 			}
 		}
 	}
+
+	int parseReplyAfter(String value) {
+		if (value == null || value.isEmpty()) {
+			return DEFAULT_RETRY_AFTER;
+		}
+
+		try {
+			int n = Integer.parseInt(value);
+			if (n < 0) {
+				return DEFAULT_RETRY_AFTER;
+			}
+
+			return Math.min(n, MAX_RETRY_AFTER);
```

Comment:
Would it make sense to model `Math.min` as sanitizer as well (unless CodeQL does that by default; which I don't think is the case)?
I think your existing `LessThanSanitizer` is incorrectly thinking that `n < 0` above is a sanitizer, which is not true. Maybe `LessThanSanitizer` could be enhanced by only holding if there is another variable access on the controlled / dominated path; or if there is no other access of the variable on the 'false' branch, e.g.:
- Not a sanitizer:
    ```java
    if (n < 0) {
        return 0;
    }
    ```
- Not a sanitizer:
    ```java
    if (n < 500) {
        return n;
    } else {
        return n / 2;
    }
    ```
- Sanitizer:
    ```java
    if (n < 500) {
        return n;
    } else {
        return 0;
    }
    ```

But maybe the maintainers can help with this (and point to existing CodeQL predicates / classes for this), or estimate whether such false negative detection is necessary.

Reply:
I've added <code>Math.min</code> as sanitizer. 

The Java API of <code>Thread.sleep</code> doesn't take a negative value of sleep time to sleep for indefinite time period. An exception will be thrown for such an input value therefore the check for negative value is not very important. The important one is the upper limit.